extension/src/lambda.js

```diff
@@ -0,0 +1,25 @@
+const utils = require('./utils')
+const paymentHandler = require('../src/paymentHandler/payment-handler')
+const setup = require('./config/init/ensure-resources')
+
+const logger = utils.getLogger()
+let init = false;
+
+exports.handler = async function(event) {
+  try {
+
+    init = init || await setup.ensureResources();
+    
+    const paymentResult = await paymentHandler.handlePayment(event.resource.obj)
+
+    return {
+      responseType: paymentResult.success ? `UpdateRequest` : `FailedValidation`,
+      errors: paymentResult.data.errors,
+      actions: paymentResult.data.actions
+    }
+  }
+  catch (e) {
+    logger.error(e, `Unexpected error when processing event ${JSON.stringify(event)}`)
```

Comment:
We were thinking that the benefit would probably be that the exception is coupled to the error message, rather than being two separate entries in cloudwatch that you have to link together manually.

Reply:
Yeah good point. The only thing I can think is if we include it there is a single entry in the logs which contains the event and the error together. But you will still have the error duplicated so not great. Happy to remove.

Reply:
I actually like that the log holds all the information together. I thought maybe we could avoid throwing after log but I guess it is not an option? I am ok if we keep it like that for now too.

Reply:
Yes I guess if it's unexpected exception that we haven't gracefully handled and we agree we want to  fail the actual lambda execution, then I guess we would have to re-throw it. So we'll leave it as is you think?

Reply:
ok